### ISSUE TITLE
Another copy of BiglyBT appears to be running

### ISSUE THREAD

**Issue by maddock34** [BiglySoftware/BiglyBT#546]
Title says it all. I try launching BiglyBT, through both the start menu and magnet links, takes a moment to appear then says ["Another copy of BiglyBT appears to be running: failed to lock file"](https://i.imgur.com/k67U7I2.png). I had this issue with Vuze but to my understanding Vuze has been effectively abandoned. 

I open torrents via magnet links, download them, remove them from BiglyBT, then close BiglyBT down. I might do this multiple times a day. 

The error seems to point to a file in the C: drive but I installed BiglyBT to the E: drive. I don't know if there's anyway of moving all the config files and what not to a different location but I doubt this would solve the issue.

I have task manager open and it does show an "active/running" BiglyBT instance but I can't end it, [tells me access is denied even in admin mode
](https://i.imgur.com/1BuVZz9.png)
OS: Windows 7 Pro SP1
BBT Version: 1.5 I think

**Comment by maddock34** [comment]
That's unfortunate. Oh well, thanks anyway

**Comment by ferdnyc** [comment]
> The error seems to point to a file in the C: drive but I installed BiglyBT to the E: drive. I don't know if there's anyway of moving all the config files and what not to a different location but I doubt this would solve the issue.

Your configuration is going to be in your `%USERPROFILEDIR%` no matter what drive the application is on, I believe in `C:\Users\<username>\AppData\Roaming\BiglyBT`. As I advised that other reporter in #554, I definitely do recommend remoiving/moving that directory (with the program not running, even in the background) and then launching BiglyBT again to let it reset the configuration from scratch. I have to believe that would fix whatever issue is causing it to get "stuck" running in the background. It's certainly worth a try.

**Comment by kryb** [comment]
I confirm having this issue as well. The .exe is still shown as running in the background, but task manager can't close it, access denied.

**Comment by linzeb** [comment]
I have this issue as well. Had this issue with Vuze also, but the fix for Vuze (adding azureus.instance.lock.disable=1 to the .proberties file) doesn't work with BiglyBT.

**Comment by parg** [comment]
That should still work - properties file is still called 'azureus.properties'

**Comment by solymosi** [comment]
I have the same issue as well. The BiglyBT.exe process just won't go away after exiting the application and it is unkillable even by Task Manager in admin mode. No other apps exhibit this issue on my system.

**Comment by solymosi** [comment]
That's definitely not it. Right now I have a "zombie" BiglyBT.exe process "running" on my system, yet I'm only using 10 GB of RAM out of 32 GB.

This is what Process Explorer reports on the zombie process:

![image](https://user-images.githubusercontent.com/199790/43678446-faa1d158-9802-11e8-91a6-eae11976fd95.png)

![image](https://user-images.githubusercontent.com/199790/43678448-09cd34f6-9803-11e8-885f-8b58ad05aa0d.png)

![image](https://user-images.githubusercontent.com/199790/43678452-1cf9bc5c-9803-11e8-9a08-43f277afb82f.png)

![image](https://user-images.githubusercontent.com/199790/43678467-44fcbdbc-9803-11e8-8ac5-f33215e8e476.png)

![image](https://user-images.githubusercontent.com/199790/43678489-aa8b8122-9803-11e8-85f9-d3163c329c79.png)

I'll try updating Java on my machine to the latest version and see if that changes anything.

**Comment by ferdnyc** [comment]
Wow, yeah, I see what you mean. It's just one stuck _thread_, and with hardly any runtime on it. (Of the process's total 8 minutes of system time and 3 minutes of kernel time, that thread accounts for... not even 100 milliseconds!) 

But it sure is jammed in there good.

**Comment by ferdnyc** [comment]
That thread start time, 8:48:43 PM on 08/03 — is that around when you initially launched BiglyBT? Or closer to when you shut it down and it went all George Romeo?

Be especially weird if it's one of the long-lived threads getting wedged on shutdown, when it hasn't really even _done_ anything the whole time.

**Comment by solymosi** [comment]
The process start time is `8:48:43 PM   2018-08-03`, so it's the former :confused: 
I'll try to live-debug the kernel following [this article](https://blogs.technet.microsoft.com/markrussinovich/2005/08/17/unkillable-processes/) and find out what's the root cause here.

**Comment by solymosi** [comment]
This looks like a stack trace of the stuck thread to me:

![image](https://user-images.githubusercontent.com/199790/43678823-b05df28c-9809-11e8-9040-34c9dc464eaa.png)

The problem seems to be occuring within `nt!NtTerminateProcess`. Let me dig a bit deeper.

**Comment by solymosi** [comment]
Seems like it's networking-related:

![image](https://user-images.githubusercontent.com/199790/43678874-67ecb852-980a-11e8-88e2-91f42a6e6b9c.png)

**Comment by solymosi** [comment]
Not sure why, but updating Java to `1.8.0_181` seems to have solved the issue :tada:

**Comment by solymosi** [comment]
Nevermind, it's broken again. I'll try to narrow down the list of other programs / services it could be conflicting with.

**Comment by maddock34** [comment]
> Installing 2018-07 Preview of Monthly Quality Rollup for Windows 7 for x64-based Systems (KB4338821) seems to fix the problem for real this time. Haven't been able to reproduce it since the update.

Yeah, since receiving this update it hasn't happened to me again either.

Don't know how to mark it as solved but it might be worthwhile keeping it around to help others that might not keep their PC up to date

**Comment by parg** [comment]
Thanks for working this one over guys! Feel free to close the issue - adding the [SOLVED] is a good idea.

**Comment by MelchiorGaspar** [comment]
ooh?! a Windows update will fix huh... ok... lol as it turns out I was putting off on that preview update...
I will try it myself... and see if it works.. hope it works for me too

**Comment by DeoW** [comment]
Just happened to me. Of interest is that while the phantom residual process was not visible on Task Manager, it was visible and could be ended in Resource Manager. Odd.
